ext/fetch/lib.rs

```diff
@@ -945,13 +978,20 @@ pub fn create_http_client(
   }
   tls_config.alpn_protocols = alpn_protocols;
 
-  let mut headers = HeaderMap::new();
-  headers.insert(USER_AGENT, user_agent.parse().unwrap());
-  let mut builder = Client::builder()
-    .redirect(Policy::none())
-    .default_headers(headers)
-    .use_preconfigured_tls(tls_config);
+  let mut http_connector = HttpConnector::new();
+  http_connector.enforce_http(false);
+  let connector = HttpsConnector::from((http_connector, tls_config));
+
+  let user_agent = user_agent
+    .parse()
+    .map_err(|_| type_error("illegal characters in User-Agent"))?;
```

Comment:
We could require the caller deals with the fact that `user_agent` should be a valid `HeaderValue`, and just pass `HeaderValue` to this function.

Reply:
Yea, this type could be pushed out to callers. FWIW, this parsing was happening right here previously, when passing the string to `reqwest::ClientBuilder::user_agent()`.